Mathlib/Algebra/Category/AlgebraCat/Monoidal.lean

```diff
@@ -0,0 +1,111 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.CategoryTheory.Monoidal.Braided
+import Mathlib.CategoryTheory.Monoidal.Transport
+import Mathlib.Algebra.Category.AlgebraCat.Basic
+import Mathlib.Algebra.Category.ModuleCat.Monoidal.Basic
+import Mathlib.RingTheory.TensorProduct
+
+/-!
+# The monoidal category structure on R-algebras
+-/
+
+open CategoryTheory
+
+universe v u
+
+variable {R : Type u} [CommRing R]
+
+namespace AlgebraCat
+
+noncomputable section
+
+namespace instMonoidalCategory
+
+open scoped TensorProduct
+
+/-- Auxiliary definition used to fight a timeout when building
+`AlgebraCat.instMonoidalCategory`. -/
+@[simps!]
+noncomputable abbrev tensorObj (X Y : AlgebraCat.{u} R) : AlgebraCat.{u} R :=
+  of R (X ⊗[R] Y)
+
+/-- Auxiliary definition used to fight a timeout when building
+`AlgebraCat.instMonoidalCategory`. -/
+noncomputable abbrev tensorHom {W X Y Z : AlgebraCat.{u} R} (f : W ⟶ X) (g : Y ⟶ Z) :
+    tensorObj W Y ⟶ tensorObj X Z :=
+  Algebra.TensorProduct.map f g
+
+/-- Auxiliary definition used to fight a timeout when building
+`AlgebraCat.instMonoidalCategory`. -/
+@[simps!]
+abbrev tensorUnit : AlgebraCat.{u} R := of R R
+
+/-- Auxiliary definition used to fight a timeout when building
+`AlgebraCat.instMonoidalCategory`. -/
+noncomputable abbrev associator (X Y Z : AlgebraCat.{u} R) :
+    tensorObj (tensorObj X Y) Z ≅ tensorObj X (tensorObj Y Z) :=
+  (Algebra.TensorProduct.assoc R X Y Z).toAlgebraIso
+
+open MonoidalCategory
+
+theorem forget₂_map_associator_hom (X Y Z : AlgebraCat.{u} R) :
+    (forget₂ (AlgebraCat R) (ModuleCat R)).map (associator X Y Z).hom
+      = (α_
+        (forget₂ _ (ModuleCat R) |>.obj X)
+        (forget₂ _ (ModuleCat R) |>.obj Y)
+        (forget₂ _ (ModuleCat R) |>.obj Z)).hom := by
+  rfl
```

Comment:
In terms of coding style, I would think we are supposed to leave the `=` sign at the end of the line. (Same for the lemma below.)

Reply:
I'm not sure we have guidance on this. Putting the operator at the beginning makes it a lot easier to find in a sea of parentheses, and is more similar to calc usage

Reply:
We have such a policy https://leanprover-community.github.io/contribute/style.html#structuring-definitions-and-theorems

Reply:
Oh, I missed the "infix operators" bit. I don't like that style, but I've changed as requested since I agree it's in there.